exercises/dolica_akelloegwel/ex02_oo_basics/src/Shape.cpp

```diff
@@ -0,0 +1,62 @@
+#include "Shape.h"
+
+Shape::Shape(std::string t, int n)
+{
+	// Set type and number of sides
+	type = t;
+	numSides = n;
+
+	// Set default member variable values
+	// Compute during subclass initialisation
+	perimeter = 0;
+	area = 0;
+	printMessage = "";
+}
+double Shape::GetPerimeter() const
+{
+	return perimeter;
+}
+void Shape::SetPerimeter(double p)

```

Comment:
In general we stay away from single letter variables in an api. However, in cases like these where it is painfully obvious it's fine.